orc/src/main/java/org/apache/iceberg/orc/OrcIterable.java

```diff
@@ -88,7 +91,7 @@ public CloseableIterator<T> iterator() {
     TypeDescription fileSchema = orcFileReader.getSchema();
     final TypeDescription readOrcSchema;
     final TypeDescription fileSchemaWithIds;
-    if (ORCSchemaUtil.hasIds(fileSchema)) {
+    if (ORCSchemaUtil.hasIds(fileSchema) && !ignoreFileFieldIds) {
```

Comment:
Nit: placing !ignoreFileFieldIds as the first condition could save the time to execute ORCSchemaUtil.hasIds() in case ignoreFileFieldIds = true

Reply:
that's a very good one, it's indeed improve the performance, will update!